vision/product_search/spec/product_search_update_product_labels_spec.rb

```diff
@@ -0,0 +1,34 @@
+# Copyright 2018 Google, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "spec_helper"
+
+describe "Update product labels" do
+
+  example "Update product labels" do
+    current_directory = File.expand_path(File.dirname(__FILE__))
```

Comment:
If every one of the specs is in the same directory, you could extract this out to the helper if you wanted to

```ruby
example "..." do
  snippet_filepath = full_snippet_path "product_search_update_product_labels.rb"

  temp_foo =
  temp_bar =
  expect(things)

  output = `ruby #{snippet_filepath} arg arg`

  expect(moar things)
end
```

or... though I know we adore not having magic, I love how explicit everything is (so maybe don't make any changes to anything)...

```ruby
example "..." do
  temp_foo =
  temp_bar =
  expect(things)

  output = run_snippet(
    "product_search_update_product_labels.rb",
    @project_id,
    @location,
    temp_product_id
  )

  expect(moar things)
end
```

🤷🏼‍♀️ Cool as is :) Optional refactor if you do/don't dig it

Reply:
Good idea! I refactored to use a helper and took out the hardcoded filenames.